src/test/java/com/cgdpd/library/service/BookServiceTest.java

```diff
@@ -86,4 +104,99 @@ public void should_throw_exception_if_author_not_exist() {
 
         verifyNoInteractions(bookRepository);
     }
+
+    @Test
+    void should_return_books_with_pagination() {
+        // given
+        var searchCriteria = SearchBookCriteria.builder().build();
+        Specification<BookEntity> spec = BookSpecifications.byBookSearchCriteria(searchCriteria);
+        int pageNumber = 0;
+        int pageSize = 10;
+        var sortParam = new SortParam("title", SortParam.Order.ASC);
+        var paginationCriteria = new PaginationCriteria(pageNumber, pageSize,
+              Optional.of(sortParam));
+        Sort sort = sortParam.toSort();
+        Pageable pageable = PageRequest.of(pageNumber, pageSize, sort);
+
+        when(bookRepository.findAll(spec, pageable))
+              .thenReturn(new PageImpl<>(List.of(
+                    JOHN_DOE__THE_ADVENTUROUS__1987,
+                    JOHN_DOE__FINDER__1995,
+                    JANE_DANE__KILLER__2001
+              )));
+
+        // when
+        PagedResponse<Book> result = bookService.getBooks(paginationCriteria, searchCriteria);
+
+        // then
+        assertThat(result.getContent().size()).isEqualTo(3);
```

Comment:
Use `assertThat(result.getContent()).containsExactlyInAnyOrder();` instead giving it the expected elements